utility/utils.py

```diff
@@ -662,3 +664,38 @@ def download_file(url, filename):
         r = requests.get(url)
         f.write(r.content)
     assert r.ok
+
+
+def get_random_str(size=13):
+    """
+    generates the random string of 13 characters
+
+    Returns:
+         str : string of random 13 characters
+
+    """
+    chars = string.ascii_lowercase + string.digits
+    return ''.join(random.choice(chars) for _ in range(size))
+
+
+def generate_data(template_name, data, template_dir="ocs-deployment"):
+    """
+    generates the data from given template (yaml)
+
+    Args:
+        template_name (str): name of the template (yaml)
+        data (dict) : data to replace in the template
+
+    Returns:
+        dict: dictionary format of data
+
+    Example:
+        generate_data("pvc.yaml", pvc_data)
+    """
+    _templating = templating.Templating()
+    template_path = os.path.join(template_dir, template_name)
```

Comment:
This implementation is very fragile.
Why not import the template to the relevant file and use it:
in openshift_ops.py import the yaml file and pass it there.
It will work?

Reply:
I am not sure w can directly import yaml and use it because we need to replace the variables in yaml file.
Other way is load yaml into dict and update dictionary with variable values which is crude way of doing it.

let me know if anyone has better way of rendering yaml file with data.

Reply:
@vavuthu Loading yaml into dict and updating dict and converting back to yaml is the clean way of edition/modifying the yaml. It will provide great flexibility inside the function to do things and not expose yaml to users at all for simplification,  users will just call the functions and pass the args that they will have to modify.

Reply:
@vasukulkarni @RazTamir I think that @vavuthu is trying to use the jinja2 templates from the reason that it's easy to have the file somewhere stored (template folder) with all variables defined where we can use powerful of jinja2 templates like filters (default and others), conditions, loops etc.  And just passing the data in the template and render it and got the data generated from template. If you will just use the pure yaml, there is no such way to use defaults, loops, conditions and so on. So I think that this is the reason why @vavuthu took this way here as we discussed this on some of our mtg.

Reply:
@petr-balogh The inital template thing is fine, but we are saying 2 things here
1) Do this operation not in utils but close to the Class function it belongs too
2) Later edit/modify/apply should be class functions rather than template modifications because it will be a small dict.

Reply:
I removed the this function since we have another function ( generate_yaml_from_jinja2_template_with_data ) which was merged recently which served the same purpose. generate_yaml_from_jinja2_template_with_data